core/src/main/java/org/apache/iceberg/FileMetadata.java

```diff
@@ -192,6 +196,15 @@ public Builder withMetrics(Metrics metrics) {
       return this;
     }
 
+    public Builder withSplitOffsets(List<Long> offsets) {
+      if (offsets != null) {
+        this.splitOffsets = copyList(offsets);
```

Comment:
I think this should use `ImmutableLists.copyOf` instead of adding a copy method just to create a mutable list.

Reply:
In that case can we move this to a util class?

Reply:
Sorry, this was my bad. I didn't realize that this `copyList` was  behavior was already present. Looks like a later change to `BaseFile` converted the offsets to `long[]` so I don't think this is actually needed anymore. I think we could use `ImmutableList.copyOf` and have the Kryo tests pass. Not a big deal though.